### ISSUE TITLE
Malformed month in IWCS 2019 .bib files

### ISSUE THREAD

**Issue by nschneid** [acl-org/acl-anthology#360]
E.g. https://aclweb.org/anthology/papers/W/W19/W19-0405.bib has

```
month = "23{--}27 " # may,
```

**Comment by mbollmann** [comment]
That looks generally correct to me, except maybe for the brackets around the dash. `may` is a BibTeX macro and `#` is string concatenation.

**Comment by nschneid** [comment]
I don't think the Zotero BibTeX parser recognizes this syntax (can't speak for other reference managers). Is this used in other Anthology .bib files?

**Comment by mbollmann** [comment]
Yes it is, see the discussion here:
https://github.com/acl-org/acl-anthology/issues/94#issuecomment-451738779

**Comment by mbollmann** [comment]
FWIW, I tried importing this entry in Zotero and it shows the date as "23--27 05 2019", which seems like it generally handles it.

**Comment by akoehn** [comment]
I can't see any reference to *days* in #94, only to multi-*month* entries.  I we wanted to have days in the bibtex entry, it should be in the `day` field, not the `month` one.

**Comment by nschneid** [comment]
> FWIW, I tried importing this entry in Zotero and it shows the date as "23--27 05 2019", which seems like it generally handles it.

For me, Zotero imports it as "23–27  2019", missing the month entirely.

In any case, the date clearly shouldn't be listed under the `month` field.

**Comment by mbollmann** [comment]
> For me, Zotero imports it as "23–27 2019", missing the month entirely.

I should have mentioned that I'm using Zotero with (Better BibTeX)[https://github.com/retorquere/zotero-better-bibtex], so maybe that's the difference?

> In any case, the date clearly shouldn't be listed under the `month` field.

Agreed. Since this is already recorded in the XML this way, I feel this is an issue with data ingestion. This particularly instance should be easy to fix, but maybe we want to make sure to check for this in the future. @mjpost?

**Comment by mjpost** [comment]
A few questions:

- Is `day` actually a bibtex field?
- How would one specify a conference date range, e.g., 30 June to 4 July?
- What does Zotero do if you import an entry with a field like `month = jan # "--" # feb`?

I think we should fix the XML. Maybe

- `month` should be `date`, which can be a range
- when we export to bib, we only export the month

**Comment by mbollmann** [comment]
`day` is not a BibTeX field, and I think representing ranges is not supported at all in plain BibTeX.

BibLaTeX, on the other hand, [supports date ranges with precision up to minutes](http://mirrors.ibiblio.org/CTAN/macros/latex/exptl/biblatex/doc/biblatex.pdf#page=16), if desired, and can also distinguish between date of publication and date of event.

**Comment by nschneid** [comment]
> What does Zotero do if you import an entry with a field like `month = jan # "--" # feb`?

For me it just imports January and ignores the rest of the line. Which I suppose is not the end of the world, if we really want to support localized month names.

If we're willing to assume English month names, any of the following will work as expected, though I realize `date` is nonstandard BibTeX:

* `month = "January--February"`
* `month = "31 January--4 February"`
* `date = "31 January--4 February, 2019"`

However, combining `date` with `month` or `year` is problematic. Exporting any of these back to BibTeX produces `month = feb`.

**Comment by mjpost** [comment]
I agree we should get rid of dates in this field, but Zotero’s failure to parse `jan # “—“ # feb` seems like a Zotero bug.
